openmc/tallies.py

```diff
@@ -822,6 +884,21 @@ def get_tally_xml(self):
             subelement = ET.SubElement(element, "nuclides")
             subelement.text = nuclides.rstrip(' ')
 
+        # Optional derivative
+        if self.diff_variable is not None:
+            if self.diff_variable == 'density':
+                subelement = ET.SubElement(element, "derivative")
+                subelement.set("variable", self.diff_variable)
+                subelement.set("material", str(self.diff_material))
+            elif self.diff_variable == 'nuclide_density':
+                subelement = ET.SubElement(element, "derivative")
+                subelement.set("variable", self.diff_variable)
+                subelement.set("material", str(self.diff_material))
+                subelement.set("nuclide", self.diff_nuclide)
+            else:
+                raise RuntimeError("Encountered unrecognized differential "
+                                   "variable in a tally.")
```

Comment:
Seems safe to omit the else block here since the setter explicitly doesn't allow values other than the two valid ones.